### ISSUE TITLE
pg_query_params and PostgreSQL keywords

### ISSUE THREAD

**Issue by mgchristensen** [php/doc-en#1716]
I have not been able to find this stated explicitly, so may I make a suggestion for the pg_query_params() documentation page?  I believe it would be helpful to state that PostgreSQL keywords such as "ASC" or "DESC" CANNOT be passed as parameters.

Thank you.

**Comment by cmb69** [comment]
I presume that you can't pass table names as parameters either. But what exactly is allowed?

Anyhow, `pg_prepare()`/`pg_execute()` likely has the same restrictions.

**Comment by mgchristensen** [comment]
Yes, it appears that table names and field names are also not allowed as parameters.  Parameters seem useful for something like a WHERE clause, as in e.g. ...WHERE table.field = $1; and the value to substitute for $1 is selected from e.g. a list box and passed into the parameter array.

**Comment by damianwadley** [comment]
Though I can't find official documentation about it (yet), presumably PostgreSQL's prepared statements work the same way as MySQL's prepared statements: placeholders can only be used with values. That excludes things like table names, column names, keywords, etc.

Because remember that the primary purpose of a prepared statement is to allow the server to _prepare_ to execute a query by having it to analyze the statement ahead of time, and without the actual data that will be involved. That means you can run the _prepared statement_ multiple times over, even with new data each time, and the server won't need to re-analyze the query each time. Placeholders for structural elements that affect the very nature of the query would preclude that sort of analysis and statement reuse.

tldr: placeholders are for data only.
